### ISSUE TITLE
Models for the Google Summer of Code 2023 prerequisite task

### ISSUE THREAD

**Issue by adrianboguszewski** [openvinotoolkit/openvino_notebooks#832]
The models listed below are for the GSoC 2023 prerequisite task. 

We provide several potential candidates. Please select only one which hasn't already been selected (look at the comments below). When you decide, assign a model to you by adding a comment with the model name. 

**Object detection:** 
Yolov6 -  https://github.com/meituan/YOLOv6 
DAMO YOLO - https://github.com/tinyvision/DAMO-YOLO 
YoloX - https://github.com/Megvii-BaseDetection/YOLOX 
RTMDet - https://github.com/open-mmlab/mmyolo/tree/main/configs/rtmdet 
EfficientDet - https://github.com/google/automl/tree/master/efficientdet 

**Rotated object detection:**
Rotated FCOS - https://github.com/open-mmlab/mmrotate/blob/main/configs/rotated_fcos/README.md 
ReDet - https://github.com/open-mmlab/mmrotate/blob/main/configs/redet/README.md 
Roi_trans - https://github.com/open-mmlab/mmrotate/blob/main/configs/roi_trans/README.md 

**Semantic Segmentation:**
SegFormer - https://huggingface.co/nvidia/segformer-b0-finetuned-ade-512-512 
ClipSeg - https://huggingface.co/CIDAS/clipseg-rd64-refined 
SETR - https://github.com/fudan-zvg/SETR 
BeIT - https://huggingface.co/microsoft/beit-base-finetuned-ade-640-640 
Segmenter - https://github.com/open-mmlab/mmsegmentation/tree/master/configs/segmenter 

**Action/Gesture recognition:** 
TSM - https://github.com/mit-han-lab/temporal-shift-module 
Timesformer - https://huggingface.co/facebook/timesformer-base-finetuned-k400 
SlowFast - https://github.com/open-mmlab/mmaction2/blob/master/configs/recognition/slowfast/README.md 

**Background matting:**
ModNet - https://github.com/ZHKKKe/MODNet 
Robust Video Background matting - https://github.com/PeterL1n/RobustVideoMatting 

**Old Photos Restoration/Image colorization/Image denoising:** 
Bringing Old Photos Back to Life - https://github.com/microsoft/Bringing-Old-Photos-Back-to-Life 
DeOldify - https://github.com/jantic/DeOldify 
Coltran - https://github.com/google-research/google-research/tree/master/coltran 
SwinIR - https://github.com/JingyunLiang/SwinIR 

**Text classification:**
Roberta - https://huggingface.co/cardiffnlp/twitter-roberta-base-sentiment 
XLM-Roberta - https://huggingface.co/papluca/xlm-roberta-base-language-detection 
DepRoBerta - https://huggingface.co/rafalposwiata/deproberta-large-depression 
CodeBerta - https://huggingface.co/huggingface/CodeBERTa-language-id 
Albert V2 - https://huggingface.co/textattack/albert-base-v2-MRPC 

**Token classification:**
Part of speech tagging - https://huggingface.co/flair/pos-english 
Punctuation restoring (bert-restore-punctuation) - https://huggingface.co/felflare/bert-restore-punctuation 
Punctuation restoring (punctuate-all) - https://huggingface.co/kredor/punctuate-all 
Typo detection - https://huggingface.co/m3hrdadfi/typo-detector-distilbert-en 
Named entity recognition - https://huggingface.co/elastic/distilbert-base-cased-finetuned-conll03-english

**Comment by sawradip** [comment]
Hello, firstly I want to thank you for the awesome list. I had a question:

Is there any existing instance/example of using `open-mmlab` models, or such config based model libraries(Ex: `detectron2`), with Openvino?

**Comment by adrianboguszewski** [comment]
@sawradip, assigned to you. We don't have such examples in this repository and I'm not aware of any existing examples of `open-mmlab` or `detectron2` with OpenVINO.

**Comment by eaidova** [comment]
@sawradip please check https://github.com/open-mmlab/mmdeploy this library support export mmlab pretrained models to OpenVINO format

**Comment by sawradip** [comment]
Thank you for assigning me that task. I will submit a PR on that, in a few days.

And I am interested on the open-mm projects is, such a good number of models, are out of openvino accessibility. Though It is not GSOC requirement, neither on planned projects, I was thinking, If I could contribute some tutorial, on integrating open-mm or similar config based models.

**Comment by akashAD98** [comment]
@eaidova should we need to implement any of the repo in openvino ir format? this is only for GSoC people or anyone from open source community can contribute

**Comment by eaidova** [comment]
@akashAD98, this activity is only for GSoC participants for now (we use this models list for candidates selection), but we are open for community contributions too :)

If you have any ideas or great models for openvino notebooks, fill free to contribute :)

**Comment by Kasliwal17** [comment]
Hi, I would like to assign SegFormer to myself.

**Comment by ahmd-nish** [comment]
Hi, i would like to assign YOLOv6  to myslef and contribute. Can you please help me with that ?

**Comment by adrianboguszewski** [comment]
Hi @ahmd-nish, please first clone the repo locally and make it run. Then please look at the 226-yolo7 notebook and create a notebook with the similar flow.

**Comment by chathuranga-jayanath-99** [comment]
Hi, I would like to assign `Roberta` to myself.

**Comment by ahmd-nish** [comment]
> Hi @ahmd-nish, please first clone the repo locally and make it run. Then please look at the 226-yolo7 notebook and create a notebook with the similar flow.

Sure I will go through and get backto you

**Comment by hazrulakmal** [comment]
Hello, I would like to work on `XLM-RoBERTa`.

**Comment by raymondlo84** [comment]
This is really cool @adrianboguszewski  -- Also, would be good to put tags in Discussion just on GSOC2023 topics to keep it organized.

**Comment by raymondlo84** [comment]
https://github.com/openvinotoolkit/openvino_notebooks/discussions/categories/gsoc-2023 :) Here we go.

**Comment by ashish-2005** [comment]
Hi , I would like to assign `EfficientDet` to myself

**Comment by adrianboguszewski** [comment]
@SpyzzVVarun XLM-Roberta has already been assigned four comments above. Please select another model :)

**Comment by SpyzzVVarun** [comment]
Then I would like to work with DepRoBerta.

**Comment by pyther-hub** [comment]
I would like to work with DAMO YOLO, My proficiency lies in computer vision.

**Comment by pyther-hub** [comment]
sir/ ma'am by what time shall we put a pr for the Prerequisite task notebook

**Comment by adrianboguszewski** [comment]
The PR must be created (doesn't have to be merged) by April 4

**Comment by Om-Doiphode** [comment]
Hi, I would like to assign `Bringing Old Photos Back to Life` to myself.

**Comment by AlexFierro9** [comment]
Hi, Would like to assign Timesformer to myself!

**Comment by Z-Fran** [comment]
Hi, I would like to assign `SwinIR` to myself.

**Comment by AnuragMaiti** [comment]
Hi, I would like to work with RTMDet and contribute! Could you please help me out with that?

**Comment by muskan-k** [comment]
Hello! I'm Muskan Kothari and I'm here to enhance my contribution to OpenVINO through GSOC'23. I'd like to assign `CodeBerta` to myself.

**Comment by dwipddalal** [comment]
Hello @adrianboguszewski 
I am Dwip Dalal. I would like to assign `Albert V2` model for Text classification to myself

**Comment by m-gopichand** [comment]
Hi , i would like to work with  ```Coltran ```  for image Colorization

**Comment by MR-ENVYR** [comment]
Hello,  @adrianboguszewski,
I am Raghavendra. I would like to assign `DistilRoberta` to myself.

**Comment by m-gopichand** [comment]
> 

Hi , i have selected ```coltrain``` , the task was to convert the model format and inference it using openvino but their are no pretrained models avaliable in the  [github](https://github.com/google-research/google-research/tree/master/coltran) directory provided , they provided script to train model but the problem is the dataset is very huge around 130GB . can i get the pre-trained model

![Screenshot_20230228_125128](https://user-images.githubusercontent.com/80762088/221782369-302a6d43-40ac-454e-8bb9-ec8965145341.png)

**Comment by Aditya-vardhan13** [comment]
Hello @adrianboguszewski ,
I am Aditya Vardhan, I would like to assign  ####Named entity recognition to myself.

**Comment by RishithaR-388** [comment]
hello @adrianboguszewski ,
I am Rishitha Reddy, I would like to assign **ClipSeg** to myself.

**Comment by hadyy17** [comment]
Hi, i would like to assign Punctuation restoring (bert-restore-punctuation) to me

**Comment by Ravindu987** [comment]
Hi, I would like to assign `Typo detection` to myself

**Comment by seanjyu** [comment]
Hi, I would like to assign the Punctuation restoring (bert-restore-punctuation) model to myself

**Comment by rajatkrishna** [comment]
Hello! I would like to contribute by working on the `SlowFast` model for video recognition.

**Comment by blaz-r** [comment]
Hi, I would like to assign `Segmenter` to myself.

**Comment by 20kavishs** [comment]
Hi, I would like to assign `TSM` to myself

**Comment by sharvesh642** [comment]
Hello @adrianboguszewski  I would like to assign  `SlowFast` to myself.

**Comment by eaidova** [comment]
Hello @sharvesh642, this task already assigned, please select another one

**Comment by sharvesh642** [comment]
Thank you for the info @eaidova, can I assign `movinet` to myself.

**Comment by Warlord-K** [comment]
Hi, can I assign gpt-neo to myself?

**Comment by rajuptvs** [comment]
Hi All, I would like to assign centernet to myself

**Comment by Abdullah-Elkasaby** [comment]
Hi!, can I assign YOLACT to myself?

**Comment by chaitravi-ce** [comment]
Hi, @maintainers could you please assign DeepLab V3 to me

**Comment by chaitravi-ce** [comment]
Hi All, could you please assign DeepLab V3 to me

**Comment by sidyakinian** [comment]
Could you assign BioGPT to me please thanks!

**Comment by 18yz153** [comment]
Could you assign CenterNet to me? Thanks you

**Comment by sahilpmehra** [comment]
Hi, could you please assign me Rotated FCOS? Thanks!

**Comment by Matrixmang0** [comment]
Hello everyone!

I would love to work on YOWOv2

Thank you

**Comment by hadyy17** [comment]
Hello, as i have assigned BeIT to myself and i am currently working on developing a notebook. I has one question, for the pre-requisite task, do i need to modify the pre-trained model mentioned with certain usecase for openvino support or i can try making a model for my own usecase using BeIT and then add required support? 

Any help is much apreciated.

**Comment by adrianboguszewski** [comment]
> Could you assign CenterNet to me? Thanks you

@18yz153, unfortunately, CenterNet model has been chosen a few comments above you. Please select another one.

**Comment by paxF3E** [comment]
Hello, I shall be willing to work with `Sound Classification`:`speech emotions recognition (wav2vec)`. Can I please have this model assigned to me?
Thank you.

**Comment by eaidova** [comment]
> Hello, as i have assigned BeIT to myself and i am currently working on developing a notebook. I has one question, for the pre-requisite task, do i need to modify the pre-trained model mentioned with certain usecase for openvino support or i can try making a model for my own usecase using BeIT and then add required support?
> 
> Any help is much apreciated.

@hadyy17 what do you mean under own use case? The most important thing that model should solve semantic segmentation task and running with openvino. If you can demonstrate own use case with preserving these conditions, it is fine

**Comment by matrixbot123** [comment]
Hi everyone !
I would like to work on  DistilBART . Thank You

**Comment by AlexFierro9** [comment]
Hi, I had assigned TimesFormer for myself, but I'm not able to convert the model to onnx correctly (some error with pytorch on my hardware), I've posted my issue on pytorch discuss but haven't gotten a response, can I change my model to a Tensorflow Based, if yes then please assign SSD MobileNet V2!

**Comment by Akshit17** [comment]
Hello everyone!
I would like to work on MiniLM. Can I have this model assigned to me?
Thank you.

**Comment by AniketARS** [comment]
Hello Everyone,
I would like to work on SETR - (https://github.com/fudan-zvg/SETR) model for my GSoc Application. I'm guessing it is unreserved, if anyone already working on this kindly inform me i will change my choice

Thanks,
Aniket

**Comment by fajemila** [comment]
I would like to assign DistilBert  to myself

**Comment by Tatwansh** [comment]
I would like to assign DETR to myself

**Comment by 18yz153** [comment]
I would like to assign YOLOR to myself

**Comment by thegeek13242** [comment]
I would like to assign YoloF to myself.

**Comment by nischay7** [comment]
Hi, I would like to assign ReDet to myself

**Comment by zhumakhan** [comment]
Mask RCNN Inception ResNet V2

**Comment by wulongjian** [comment]
Hi, I would like to assign Roi_trans to myself. Thanks.

**Comment by AlexFierro9** [comment]
Hi, I've an already built copy of SSD NET V2 at [here at TF hub]( https://storage.googleapis.com/tfhub-modules/tensorflow/ssd_mobilenet_v2/fpnlite_320x320/1.tar.gz), and while the model can be rebuilt again from this [checkpoints](http://download.tensorflow.org/models/object_detection/tf2/20200711/ssd_mobilenet_v2_320x320_coco17_tpu-8.tar.gz) too, This way also requires using Tensor Object Detection API that needs to be separately installed, to be able to recompile the model and save it. So should I use the already existing saved model or build it again?

**Comment by eaidova** [comment]
@AlexFierro9 you can use any of this model, but please pay attention that it is 2 different models, the first one in Tf Object detection API named as [SSD MobileNet V2 FPNLite 320x320](http://download.tensorflow.org/models/object_detection/tf2/20200711/ssd_mobilenet_v2_fpnlite_320x320_coco17_tpu-8.tar.gz)

**Comment by hazrulakmal** [comment]
Hi, this might seem like a dumb question. Does one need a machine with an Intel processor to meaningfully contribute to this organization and complete the prerequisite task? Would someone with, say, a Ryzen processor be able to complete the task?

**Comment by pyther-hub** [comment]
@eaidova @adrianboguszewski I would like to change the model that was previously assigned to me as there are many in consistencies in the DAMO YOLO, I would like to use YoloV7 Face

**Comment by mhy-666** [comment]
Hi, can I assign Deberta to myself?
Thanks.

**Comment by pyther-hub** [comment]
@eaidova sorry for the inconvenience but assign me  NanoDet as there is no active documentation for yolo v7 face

**Comment by sususama** [comment]
Hi, I want to assign  Photo2Cartoon.Thanks.

**Comment by JacketChenlll** [comment]
Hi, could you please assign me UltraFace? If I work on this successfully before March 20th, I will also try DAMO YOLO, Thanks!

**Comment by Nouran-Muhammad** [comment]
Hi, Could you please assign me ESPNet? Thank you

**Comment by weronikazak** [comment]
Hello, could you please assign me Coltran? Thank you!

**Comment by wulongjian** [comment]
Hello, I had assigned Roi_trans for myself, but I'm not able to convert it correctly in intel cpu platform. As far as I konw, this model only support nvidia inference currently(see this[ issue#551](https://github.com/open-mmlab/mmrotate/issues/511#issuecomment-1253375395) and [get_started](https://github.com/open-mmlab/mmrotate/blob/main/docs/en/get_started.md#test-a-model)).   
Can I change my model to others?  if yes then please assign Robust Video Background matting to me, thanks a lot.

**Comment by AlexFierro9** [comment]
Hi,
So I had a doubt with model conversion, I've been refering to the model conversion guide here [Converting TensorFlow Object Detection API Models](https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_convert_model_tf_specific_Convert_Object_Detection_API_Models.html) as according to the documentation and colab tutorials available [here](https://github.com/tensorflow/models/blob/master/research/object_detection/colab_tutorials/inference_tf2_colab.ipynb).
The mo command for conversion requires me to also specify **--transformations_config <path_to_subgraph_replacement_configuration_file.json>** for argument, unfortunately this model was not downloaded from tensorflow object detection api zoo, so I have no clue which file to use here, should I outright skip this parameter or should I use the json file specified for ssd_v2? btw my model is ssd_mobilenet_v2_320x320_coco17_tpu-8

**Comment by shrey-2803** [comment]
Hi,
I would like to assign FinBERT to myself.
Thank You!

**Comment by sharvesh642** [comment]
Hi @adrianboguszewski @eaidova ,
I was assigned with the `movinet` task and while I was trying to load the tensorflow model after converting it into OPENVINO IR I encountered an error - "Model file model/v3-small_224_1.0_float.xml cannot be opened!"
When I was trying to Load the model in OpenVINO Runtime with ie.read_model I encountered the error - "Model file model/MiDaS_small.xml cannot be opened!".
Am I doing something wrong how can I rectify it?

**Comment by aadhamm** [comment]
Hi, @adrianboguszewski @eaidova 

I would like to assign "Real-ESRGAN (for real images)" to myself.
Thank you for this insightful list and best of luck for all of us.

**Comment by eaidova** [comment]
@sharvesh642 which platform do you use for development? Please also check that your path to model is correct and both IR files (XML and BIN )are located in the same directory and have the same name without extension

**Comment by eaidova** [comment]
> Hi, So I had a doubt with model conversion, I've been referring to the model conversion guide here [Converting TensorFlow Object Detection API Models](https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_convert_model_tf_specific_Convert_Object_Detection_API_Models.html) as according to the documentation and Colab tutorials available [here](https://github.com/tensorflow/models/blob/master/research/object_detection/colab_tutorials/inference_tf2_colab.ipynb). The mo command for conversion requires me to also specify **--transformations_config <path_to_subgraph_replacement_configuration_file.json>** for argument, unfortunately this model was not downloaded from tensorflow object detection api zoo, so I have no clue which file to use here, should I outright skip this parameter or should I use the json file specified for ssd_v2? btw my model is ssd_mobilenet_v2_320x320_coco17_tpu-8

@AlexFierro9 you can use any suitable SSD mobilenet v2 model which you can run and convert, so model selection should not be a problem, right? :)

TF object detection models saved using in saved model format correspond to TF2.0 and should work without pipeline config

**Comment by pyther-hub** [comment]
@eaidova My model is nanodet in that when I compile my model 
```
from openvino.runtime import Core
core = Core()
model = core.read_model('/content/nanodet_model.xml')
compiled_model = core.compile_model(model, 'CPU')
```
after this 
```
output_blob = compiled_model.output(0)
predictions = torch.from_numpy(compiled_model(input_tensor)[output_blob])
```
the shape of my predictions is 2125, 112 where 2125 is number of boxes 
now the main problem comes up of 112, ideally it should be 85 
1+4+80 (one for probability of an object, 4 is the coordinate of the box, 80 is the prob. of each class)

after doing a ton of dugging I found out that the model returns this only after which it needs to be passed through [post_process](https://github.com/RangiLyu/nanodet/blob/0b78ebabfaec0edabb4e6ca629fc51130f1505d5/nanodet/model/head/nanodet_plus_head.py#L464)
where it decodes the bboxes and all the functions are complex and inter linked I am not able to move from here like how can I get the required output from this 112

**Comment by MR-ENVYR** [comment]
Hello,
My model is distilroberta. There is no tiny model for it so should I just use the model file available and write an export script for the model? I had a doubt as the example notebook used an export script which was available in the github repo of the model. So should I follow a similar approach of the export script or can I use any approach as convenient?

**Comment by eaidova** [comment]
> Hello, My model is distilroberta. There is no tiny model for it so should I just use the model file available and write an export script for the model? I had a doubt as the example notebook used an export script which was available in the github repo of the model. So should I follow a similar approach of the export script or can I use any approach as convenient? Thanks in advance :D

you can use any approach which can be suitable

**Comment by eaidova** [comment]
> @eaidova My model is nanodet in that when I compile my model
> 
> ```
> from openvino.runtime import Core
> core = Core()
> model = core.read_model('/content/nanodet_model.xml')
> compiled_model = core.compile_model(model, 'CPU')
> ```
> 
> after this
> 
> ```
> output_blob = compiled_model.output(0)
> predictions = torch.from_numpy(compiled_model(input_tensor)[output_blob])
> ```
> 
> the shape of my predictions is 2125, 112 where 2125 is number of boxes now the main problem comes up of 112, ideally it should be 85 1+4+80 (one for probability of an object, 4 is the coordinate of the box, 80 is the prob. of each class)
> 
> after doing a ton of dugging I found out that the model returns this only after which it needs to be passed through [post_process](https://github.com/RangiLyu/nanodet/blob/0b78ebabfaec0edabb4e6ca629fc51130f1505d5/nanodet/model/head/nanodet_plus_head.py#L464) where it decodes the bboxes and all the functions are complex and inter linked I am not able to move from here like how can I get the required output from this 112

Possibly, I do not quite understand what the issue is, what is the problem to apply post-processing to model output after openvino inference?   It is normal practice that some models can require additional steps for getting results.  e.g. if you look on yolov7 tutorial it is also requires to make non-maximum suppression for boxes after inference

**Comment by mhy-666** [comment]
Hi, can I assign ELECTRA to myself?
Thanks.

**Comment by pyther-hub** [comment]
> > @eaidova My model is nanodet in that when I compile my model
> > ```
> > from openvino.runtime import Core
> > core = Core()
> > model = core.read_model('/content/nanodet_model.xml')
> > compiled_model = core.compile_model(model, 'CPU')
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > after this
> > ```
> > output_blob = compiled_model.output(0)
> > predictions = torch.from_numpy(compiled_model(input_tensor)[output_blob])
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > the shape of my predictions is 2125, 112 where 2125 is number of boxes now the main problem comes up of 112, ideally it should be 85 1+4+80 (one for probability of an object, 4 is the coordinate of the box, 80 is the prob. of each class)
> > after doing a ton of dugging I found out that the model returns this only after which it needs to be passed through [post_process](https://github.com/RangiLyu/nanodet/blob/0b78ebabfaec0edabb4e6ca629fc51130f1505d5/nanodet/model/head/nanodet_plus_head.py#L464) where it decodes the bboxes and all the functions are complex and inter linked I am not able to move from here like how can I get the required output from this 112
> 
> @pyther-hub Possibly, I do not quite understand what the issue is, what is the problem to apply post-processing to model output after openvino inference? It is normal practice that some models can require additional steps for getting results. e.g. if you look on yolov7 tutorial it is also requires to make non-maximum suppression for boxes after inference

ma'am the issue is the object returns a diff. output and it needs to be post processed but the post processing step is very very complex, I am trying to. implement that I have one more backup option which I will look later I would keep you posted regarding my progress, thank you for your response

**Comment by pyther-hub** [comment]
@eaidova I would like to change my model from nanodet to Colorizer
I apologize for any inconvenience I may have caused. Despite my best efforts, I encountered difficulties in resolving the output issue. I reached out to several contributors via email, but one of them informed me that the code required for this conversion would be quite challenging to write. Moreover, the output I obtained did not match the model.forward result, which ultimately led me to a dead end.

**Comment by mhy-666** [comment]
> Hi, can I assign ELECTRA to myself? Thanks.

I'm sorry, I didn't notice the requirement that only one task can be assigned to a person. I have already completed the task that I previously received.

**Comment by adrianboguszewski** [comment]
> @eaidova I would like to change my model from nanodet to Colorizer I apologize for any inconvenience I may have caused. Despite my best efforts, I encountered difficulties in resolving the output issue. I reached out to several contributors via email, but one of them informed me that the code required for this conversion would be quite challenging to write. Moreover, the output I obtained did not match the model.forward result, which ultimately led me to a dead end.

It's ok. Reassigned.

**Comment by adrianboguszewski** [comment]
> > Hi, can I assign ELECTRA to myself? Thanks.
> 
> I'm sorry, I didn't notice the requirement that only one task can be assigned to a person. I have already completed the task that I previously received.

Yes. One notebook implementation is enough. Please be sure you will have opened a PR (it's part of the task) with your work by 4th April.

**Comment by pyther-hub** [comment]
@adrianboguszewski sir I have made the notebook for coloriser task but while doing documentation I found that one [implementation of it is already there] (https://github.com/openvinotoolkit/openvino_notebooks/tree/main/notebooks/222-vision-image-colorization) , what shall I do now?

**Comment by adrianboguszewski** [comment]
> @adrianboguszewski sir I have made the notebook for coloriser task but while doing documentation I found that one [implementation of it is already there] (https://github.com/openvinotoolkit/openvino_notebooks/tree/main/notebooks/222-vision-image-colorization) , what shall I do now?

Just create a PR. If it was listed above, it means we would like to have it anyway. When you create a PR, we will let you know about the next steps.

**Comment by lucifertrj** [comment]
Hi @adrianboguszewski , 

Please assign `YoloV7 Face` to me

**Comment by AlexFierro9** [comment]
#923

**Comment by AlexFierro9** [comment]
#923

**Comment by Muskan33** [comment]
Hi @adrianboguszewski ,

Please assign (https://github.com/yulunzhang/RCAN) to me

**Comment by Tselmuunkoko** [comment]
Hello, Could you assign me ModNet? Thank you very much.

**Comment by Aneesh02** [comment]
Hi @adrianboguszewski ,
Please assign [NanoDet](https://github.com/RangiLyu/nanodet) to me. Looks like people have have already tried to do it still I will give my best.

**Comment by AlexFierro9** [comment]
#923 
I've met the self-checks, can someone review the PR?

**Comment by adrianboguszewski** [comment]
> #923 I've met the self-checks, can someone review the PR?

Yes, we will review it when we find some resources for that. We process PRs in FIFO style. So first, we will review all PRs created before this one. 

The PR must be opened before April 4th, it doesn't have to be reviewed or merged.

**Comment by sanjayk0508** [comment]
Hey, @adrianboguszewski. I'm Sanjay, you can assign me Question Answering - ELECTRA

**Comment by evawyf** [comment]
Hi @adrianboguszewski , could you assign the `Mask RCNN Inception ResNet V2` to me. Thanks!

**Comment by evawyf** [comment]
Hello @adrianboguszewski , could you assign the `Mask RCNN Inception ResNet V2` to me? Thanks!

**Comment by BrennoMello** [comment]
Hi @adrianboguszewski., please assign **Timesformer** (https://huggingface.co/facebook/timesformer-base-finetuned-k400) to me.

**Comment by mr-rajashekhar** [comment]
Hey @adrianboguszewski, please assign Mask RCNN Inception ResNet V2 to me.

**Comment by zhumakhan** [comment]
Hi all,
my model is OPT, and it is quantized successfully to obtain better speed in PR  #943. However, 2 tests are failing due to timeout because the model is quite large. Will this fact be considered during the review? If not, can I assign myself to ModNet?

Thanks!

**Comment by adrianboguszewski** [comment]
> Hi all, my model is OPT, and it is quantized successfully to obtain better speed in PR #943. However, 2 tests are failing due to timeout because the model is quite large. Will this fact be considered during the review? If not, can I assign myself to ModNet?
> 
> Thanks!

Yes, we are aware that huge models cannot finish processing in a specified time. We accept your PR as a prerequisite task :)

**Comment by adrianboguszewski** [comment]
**Dear contributors, if you decide not to finish your prerequisite task before April 4th, please kindly tell us about it asap. We want to release not actively developed models to others.**

**Comment by pyther-hub** [comment]
sir do we need to do quantisation also for this task? as my model is colorisation only and for the pre training the dataloader is very different and even I do not have any quantitative method for calculating how good my model is it just converts black n white to color image

**Comment by chathuranga-jayanath-99** [comment]
Hi i also like to release.

**Comment by Aneesh02** [comment]
> **Dear contributors, if you decide not to finish your prerequisite task before April 4th, please kindly tell us about it asap. We want to release not actively developed models to others. Thank you for that!**

Hi @adrianboguszewski , I would like to release NanoDet. Apologies for any inconvenience caused.

**Comment by Nouran-Muhammad** [comment]
Hello @adrianboguszewski ,
I have been working on the prerequisite task on ESPNet model. I created an openvino environment and installed all the requirements according to anaconda installation instructions [here](https://github.com/openvinotoolkit/openvino_notebooks/wiki/Conda). I cloned the model's repo and I have been trying to run inference using the pytorch version of the model or even load the model. It gives me an error saying that **torch.cuda.is_available() is False**.  CUDA cannot be detected although it can be detected in the base environment and any other environment I have created before. I have tried a lot with different versions but that would result in dependencies conflict.
I have uninstalled and reinstalled openvino in case any packages didn't install correctly but it didn't work.
what should i do?
Thanks

**Comment by pyther-hub** [comment]
@adrianboguszewski @eaidova do we need to put quantisation part also in our notebook?

**Comment by AnuragTimilsina** [comment]
Hello, I want to assign `yolov5-blazeface` to myself.

**Comment by 20kavishs** [comment]
Hi, I would like to release please.

**Comment by nsk126** [comment]
Hi, I would like to assign `MiDaS 3.1` to mysef.

**Comment by adrianboguszewski** [comment]
> sir do we need to do quantisation also for this task? as my model is colorisation only and for the pre training the dataloader is very different and even I do not have any quantitative method for calculating how good my model is it just converts black n white to color image

No, the quantization part is not mandatory

**Comment by VaillaRohit** [comment]
can you please assign RetinaFace as prerequisite task to me?

**Comment by Nouran-Muhammad** [comment]
Hi @adrianboguszewski,
i have a question. is it a must that my pull request be accepted to be able to submit a proposal and apply for a project and be considered in the filtration phase by you?

**Comment by adrianboguszewski** [comment]
> Hi @adrianboguszewski, i have a question. is it a must that my pull request be accepted to be able to submit a proposal and apply for a project and be considered in the filtration phase by you?

You have to provide a link to your PR in your application. It means the due date for that is April 4th (as you cannot upload a new version of the proposal after that date). Once again, the PR must be open; it doesn't have to be reviewed or merged.

**Comment by zilto** [comment]
Hi! I would like to implement CodeBerta to complete the prerequesite for GSoC 2023!

**Comment by ABHIJATSARARI** [comment]
I would like to assign Roberta to myself.

**Comment by mhy-666** [comment]
Hello, I have completed all the requirements requested in my PR, but I am unable to tick the checkbox items. Could you please review my PR?

**Comment by matrixbot123** [comment]
Hi, I would like to release please.

**Comment by BrennoMello** [comment]
Hello @adrianboguszewski , is it a problem not using the activity's repository?  
My model timesformer specifies this repository (https://huggingface.co/facebook/timesformer-base-finetuned-k400), but huggingface API does not export the model timesformer-base-finetuned-k400 to ONNX. So, I started to use this repository (https://github.com/wanchichen/TimeSformer), a fork from the official with new corrections.

**Comment by adrianboguszewski** [comment]
> Hello, I have completed all the requirements requested in my PR, but I am unable to tick the checkbox items. Could you please review my PR?

You can copy a list and post it yourself. Then you're able to tick.

**Comment by adrianboguszewski** [comment]
> Hi @adrianboguszewski It would be great if `YoLoP` for semantic segmentation could be assigned to me as a pre-requisite for GSoC'23. Additionally, it would be great if some SOTA `YoLov8` object detection models from the Ultralytics [repo](https://github.com/ultralytics/ultralytics) could be included in object detection — just a small suggestion from my side.

It was already done a few months ago. Check notebook 230.

**Comment by adrianboguszewski** [comment]
> Hello @adrianboguszewski , is it a problem not using the activity's repository? My model timesformer specifies this repository (https://huggingface.co/facebook/timesformer-base-finetuned-k400), but huggingface API does not export the model timesformer-base-finetuned-k400 to ONNX. So, I started to use this repository (https://github.com/wanchichen/TimeSformer), a fork from the official with new corrections.

It's ok to clone another repository, but it should be avoided if more straightforward methods to get a model are available.

**Comment by samycolen** [comment]
Hello there ........
Would like to assign DistilBART to myself!

**Comment by ntombi** [comment]
Hi. I would like to assign TSM to myself. Thanks.

**Comment by pronoym99** [comment]
Hi, @adrianboguszewski I would like to **relieve** myself of YoLop and would instead assign myself [VI-Depth](https://github.com/isl-org/VI-Depth). Sorry for any inconvenience caused.

**Comment by sahilpmehra** [comment]
Hi, @adrianboguszewski I am having trouble converting the Rotated FCOS pytorch model to onnx format due to certain support issues. I am sure there is a way around it but due to the deadline coming up soon, I would like to release Rotated FCOS and instead assign NanoDet to myself. Apologies for the inconvenience.

**Comment by iamaminebriki** [comment]
I would like to assign `NanoDet` to myself!

**Comment by Lj1ang** [comment]
I'd like to assign  `FaceParsing |& MakeUp` to myself

**Comment by MR-ENVYR** [comment]
Hello, Can we install any other library using !pip install inside the notebook for our specific case or do we need to only make do with the shared libraries of requirements.txt? I wish to use the transformers and optimum library library here.
Thanks in advance :D

**Comment by samycolen** [comment]
hello @adrianboguszewski where do i have to  submit the PR ? is it in My Proposal?

**Comment by 18yz153** [comment]
Hello, I am working on YoloR. The weight and dataset are downloaded by script. It is sh scripts. It does not work on my Windows. But it works on colab. I tried to use notebook_utils download_file. It tells me `24 from IPython.display import HTML, Image, Markdown, display
     25 from matplotlib.lines import Line2D
---> 26 from models import model
     27 from tqdm.notebook import tqdm_notebook
     30 # ## Files
     31 # 
     32 # Load an image, download a file, download an IR model, and create a progress bar to show download progress.
     33 
     34 # In[ ]:

ImportError: cannot import name 'model' from 'models' `
How am I supposed to download weights and datasets? I already convert the model into openvino. It works on inference.

**Comment by VaillaRohit** [comment]
Hello, I have completed the prerequisite task.
PR-GSOC-2023 Retinaface by @VaillaRohit #979

**Comment by MR-ENVYR** [comment]
> Hello, I am working on YoloR. The weight and dataset are downloaded by script. It is sh scripts. It does not work on my Windows. But it works on colab. I tried to use notebook_utils download_file. It can not download google drive files. How am I supposed to download weights and datasets? I already convert the model into openvino. It works on inference.

Try to change your google drive link using [this website](https://paperform.co/templates/apps/direct-download-link-google-drive/). Passing this link to the notebook_utils `download_file` function worked for me.  🤠

**Comment by 18yz153** [comment]
> > Hello, I am working on YoloR. The weight and dataset are downloaded by script. It is sh scripts. It does not work on my Windows. But it works on colab. I tried to use notebook_utils download_file. It can not download google drive files. How am I supposed to download weights and datasets? I already convert the model into openvino. It works on inference.
> 
> Try to change your google drive link using [this website](https://paperform.co/templates/apps/direct-download-link-google-drive/). Passing this link to the notebook_utils `download_file` function worked for me. 🤠

I download nothing. Google Drive can't scan this file for viruses.

[yolor_p6.pt](https://drive.google.com/open?id=1Tdn3yqpZ79X7R1Ql0zNlNScB1Dv9Fp76&authuser=0) (143M) is too large for Google to scan for viruses. Would you still like to download this file?
I guess this is the problem. Google ask me to click download again. Anyone know how to fix this?

**Comment by wonhyeongseo** [comment]
Dear @adrianboguszewski ,

I recently discovered GSoC and I am interested in contributing to the project by working on `Rotated object detection > Roi_trans`. I understand that the deadline is today, but I would still like to give it my best shot. Thank you for considering my request.

Best regards,
Wonhyeong Seo

**Comment by adrianboguszewski** [comment]
> Hello, Can we install any other library using !pip install inside the notebook for our specific case or do we need to only make do with the shared libraries of requirements.txt? I wish to use the transformers and optimum library library here. Thanks in advance :D

@MR-ENVYR, yes it's ok to use pip in a notebook. It has been done like that in e.g. 230-yolov8

**Comment by adrianboguszewski** [comment]
> > > Hello, I am working on YoloR. The weight and dataset are downloaded by script. It is sh scripts. It does not work on my Windows. But it works on colab. I tried to use notebook_utils download_file. It can not download google drive files. How am I supposed to download weights and datasets? I already convert the model into openvino. It works on inference.
> > 
> > 
> > Try to change your google drive link using [this website](https://paperform.co/templates/apps/direct-download-link-google-drive/). Passing this link to the notebook_utils `download_file` function worked for me. 🤠
> 
> I download nothing. Google Drive can't scan this file for viruses.
> 
> [yolor_p6.pt](https://drive.google.com/open?id=1Tdn3yqpZ79X7R1Ql0zNlNScB1Dv9Fp76&authuser=0) (143M) is too large for Google to scan for viruses. Would you still like to download this file? I guess this is the problem. Google ask me to click download again. Anyone know how to fix this?

@eaidova, do you know how to deal with that issue?

**Comment by sawradip** [comment]
> > > > Hello, I am working on YoloR. The weight and dataset are downloaded by script. It is sh scripts. It does not work on my Windows. But it works on colab. I tried to use notebook_utils download_file. It can not download google drive files. How am I supposed to download weights and datasets? I already convert the model into openvino. It works on inference.
> > > 
> > > 
> > > Try to change your google drive link using [this website](https://paperform.co/templates/apps/direct-download-link-google-drive/). Passing this link to the notebook_utils `download_file` function worked for me. 🤠
> > 
> > 
> > I download nothing. Google Drive can't scan this file for viruses.
> > [yolor_p6.pt](https://drive.google.com/open?id=1Tdn3yqpZ79X7R1Ql0zNlNScB1Dv9Fp76&authuser=0) (143M) is too large for Google to scan for viruses. Would you still like to download this file? I guess this is the problem. Google ask me to click download again. Anyone know how to fix this?
> 
> @eaidova, do you know how to deal with that issue?

I think I can help. 

You can use [`gdown`](https://github.com/wkentaro/gdown).  it works for all OS. Just extract the file id from the google drive link and use `gdown <drive-id>`. For larger files you can use: `gdown "<drive-id>&confirm=t"`. Got from [here](https://stackoverflow.com/questions/60739653/gdown-is-giving-permission-error-for-particular-file-although-it-is-opening-up-f)

It also has a python API where you can `import gdown` in pythonic way (look in the linked Repo). Hope this helps!

**Comment by adrianboguszewski** [comment]
As the 4th of April has passed, this task is suspended until further notice. Thank you all for your contribution :)

**Comment by raymondlo84** [comment]
Thank you everybody <3
